li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/consumer/LiKafkaConsumerImpl.java

```diff
@@ -461,6 +484,42 @@ private long positionMain(TopicPartition partition, Duration timeout) {
     }
   }
 
+  private ConsumerRecordsProcessingException handleRecordProcessingException(TopicPartition tp) {
+    if (_lastProcessedResult != null && _lastProcessedResult.exception() != null) {
+      // if tp is null handle all exception otherwise handle exception for given tp
+      if (tp == null || _lastProcessedResult.hasError(tp)) {
+        seekToResumeOffsetsOnRecordProcessingExceptions();
+        ConsumerRecordsProcessingException e = _lastProcessedResult.exception();
+        _lastProcessedResult = null;
+        return e;
+      }
+
+    }
+    return null;
+  }
+
+  private void seekToResumeOffsetsOnRecordProcessingExceptions() {
```

Comment:
any other invocations of this method except the above one? if not the condition check could be avoided